aten/src/ATen/native/Memory.cpp

```diff
@@ -10,8 +10,8 @@ Tensor pin_memory(const Tensor& self) {
   if (self.type().backend() != kCPU) {
     AT_ERROR("cannot pin '", self.type().toString(), "' only CPU memory can be pinned");
   }
-  auto allocator = detail::getCUDAHooks().newPinnedMemoryAllocator();
-  auto tensor = self.type().tensorWithAllocator(self.sizes(), self.strides(), std::move(allocator));
+  auto* allocator = detail::getCUDAHooks().getPinnedMemoryAllocator();
```

Comment:
why the `auto*`?  This would break if we turn the PinnedMemoryAllocator into some non-pointer type right?  Is that what you want?

Reply:
Yeah. The point is to force you to look at the call site and fix it, if necessary.

Reply:
So don't use auto at all? :).  It's fine, I just don't get the half way choice.